### PR TITLE
[#1318] Actual 1/2: Enable “Scan QR” button only if camera available

### LINKED ISSUE #1318: Disable "Scan QR Code" button if camera is not available while logging in with private key
**Wallet version or commit SHA:**
`dev` branch (ae58feb)

**Environment:**
Win 10 Home (64 bit) Version 1803 Build 17134.112

**Pre-requisites:**
No camera available (VM for example)

**Reproduction steps:**
1. Start Neon Wallet
2. Click on "Login with private key" button

If there is no camera available when going to "Login with private key" page, "Scan QR Code" button should be disabled. Currently if button is clicked, there's an error thrown in console however it's not visible in the wallet UI, so users are not informed about this. Also button changes to "Cancel" - it shouldn't happen in case of error.

If it's not possible to check camera and disable this button, we should at least show the error after clicking on it.

**Screenshot/Video:**
show_error

### PR DESCRIPTION
**Pull request by ranbena**
Fixes #1318.

**tl;dr** The actual feature work

NOTICE: This PR depends on #1730 and #1731 to be merged so defer review till then plz.

#### The #1318 Pull Request stack
1. (#1730) PrepWork 1/3: Added util/poll.js unit tests
2. (#1731) PrepWork 2/3: Made util/poll.js cancellable with cancellablePoll
3. (#1731) PrepWork 3/3: Added cancellablePoll unit tests
4. (#1732) Actual 1/2: Enable “Scan QR” button only if camera available  ☜☜☜ _**YOU ARE HERE**_
5. (#1733) Actual 2/2: Added camera availability unit tests

#### What's this commit about?
The main feature finally. Basically, once the Private Key Login tab is clicked, the "Scan QR" button isn't enabled until a camera source is found (not "active" but rather "available" in the OS).

This enhancement to `LoginPrivateKey` is implemented with an HOC `withCameraAvailability` which adds a prop `cameraAvailable` to the wrapped component and automatically toggles the boolean value according to actual camera availability.

The availability is determined depending on [`navigator.mediaDevices.enumerateDevices()`](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/enumerateDevices). This required updating Flow's BOM declarations (this browser feature isn't yet supported in IE 😪...)

The listener is implemented with the `util/poll.js` polling mechanism. When the state of availability changes (for instance, a usb camera was connected), the polling ends, and another is set listening to the opposite value (when the camera is disconnected). Eventually, polling is stopped when component is unmounted.

LETS SEE IT IN ACTION
ezgif com-video-to-gif

### CODE REVIEW

File: app/hocs/withCameraAvailability.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+const POLL_FREQUENCY = 1000
+
+type State = {
+  avail: boolean
+}
+
+type Props = {}
+
+// $FlowFixMe
+export default function withCameraAvailability(Component) {
```
Comment (comountainclimber) [inline app/hocs/withCameraAvailability.js]: Awesome use case for a HOC 👍
